indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -3977,14 +4048,25 @@ protected void emitNoticesQueueSize()
 
   protected void emitLag()
   {
-    if (spec.isSuspended() || !stateManager.isSteadyState()) {
+    log.info(stateManager.getSupervisorState().getBasicState().toString());
+    log.info("!isIdle: %s, spec.isSuspended: %s, !statemanager.isSteadyState %s", !isIdle(), spec.isSuspended(), !stateManager.isSteadyState());
+    if (spec.isSuspended() || !stateManager.isSteadyState() && !isIdle()) {
       // don't emit metrics if supervisor is suspended or not in a healthy running state
       // (lag should still available in status report)
       return;
     }
     try {
-      Map<PartitionIdType, Long> partitionRecordLags = getPartitionRecordLag();
-      Map<PartitionIdType, Long> partitionTimeLags = getPartitionTimeLag();
+      Map<PartitionIdType, SequenceOffsetType> offsetsFromMetadatStorage = getOffsetsFromMetadataStorage();
```

Comment:
These changes can be reverted if the metadata storage offsets are returned instead of any empty map when there are no active tasks (in getHighestCurrentOffsets)